junit-platform-engine/src/main/java/org/junit/platform/engine/support/descriptor/DefaultUriSource.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2015-2018 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v2.0 which
+ * accompanies this distribution and is available at
+ *
+ * http://www.eclipse.org/legal/epl-v20.html
+ */
+
+package org.junit.platform.engine.support.descriptor;
+
+import static org.apiguardian.api.API.Status.STABLE;
+
+import java.net.URI;
+import java.util.Objects;
+
+import org.apiguardian.api.API;
+import org.junit.platform.commons.util.Preconditions;
+import org.junit.platform.commons.util.ToStringBuilder;
+
+/**
+ * Default uri-based test source implementation.
+ *
+ * @since 1.3
+ */
+@API(status = STABLE, since = "1.3")
+class DefaultUriSource implements UriSource {
+
+	private static final long serialVersionUID = 1L;
+
+	private final URI uri;
+
+	DefaultUriSource(URI uri) {
+		this.uri = Preconditions.notNull(uri, "uri must not be null");
```

Comment:
uri must not be null -> URI must not be null

Reply:
✔️

Reply:
There's wild mix of lower- and upper-case vs type and parameter name usages of `Preconditions::<T> T notNull(T object, String message)`... could need polishing pass.

Reply:
sure.... I imagine I'll take another look in an actual browser once it's all in master.